### PR TITLE
Add the rDG module

### PR DESCRIPTION
**Pull request by permcody**
refs #7967

### CODE REVIEW

**Comment by permcody** [comment]
Ping @yidongxiainl

File: modules/rdg/src/base/RdgApp.C [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#include "MooseSyntax.h"
+
+template<>
+InputParameters validParams<RdgApp>()
+{
+  InputParameters params = validParams<MooseApp>();
+
+  params.set<bool>("use_legacy_uo_initialization") = false;
+  params.set<bool>("use_legacy_uo_aux_computation") = false;
+  params.set<bool>("use_legacy_output_syntax") = false;
```
Comment (andrsd) [inline modules/rdg/src/base/RdgApp.C]: I would be nice to have these default to `false`, so new stuff does not have to mess with historical garbage...

File: modules/rdg/src/main.C [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+#include "MooseInit.h"
+#include "Moose.h"
+#include "MooseApp.h"
+#include "AppFactory.h"
+
+// Create a performance log
+PerfLog Moose::perf_log("Rdg");
+
+// Begin the main program.
+int main(int argc, char *argv[])
```
Comment (andrsd) [inline modules/rdg/src/main.C]: Strictly speaking `int` should be on a separate line ;-) I guess we just did not modify `stork` yet...

**Review by andrsd — APPROVED** [review]
I do not think we need to address the default values.

And I do not really care for the `int main` line. I do not believe any other app does that, so good with me. Maybe the automated formatter will fix this eventually...

**Comment by permcody** [comment]
Switching the defaults can probably be done... this means verifying that
all of application that are using legacy options are setting them
explicitly. That'll take little time and probably shouldn't hold this up.

**Comment by andrsd** [comment]
> Switching the defaults can probably be done...

I think so too, we did that switch a long time ago. And everybody should be using the new behavior... If they need the old one, they should have these 3 lines in their apps, not the other way around...

**Pull request merged**